helpers/env-variables.ts

```diff
@@ -151,21 +151,55 @@ const getModelEnvs = (modelConfig: ModelConfig): EnvVar[] => {
       description: "Name of the embedding model to use.",
       value: modelConfig.embeddingModel,
     },
-    {
-      name: "EMBEDDING_DIM",
-      description: "Dimension of the embedding model to use.",
-      value: modelConfig.dimensions.toString(),
-    },
-    ...(modelConfig.provider === "openai"
-      ? [
+  ];
+
+  switch (modelConfig.provider) {
+    case "openai": {
+      modelEnvs.push(
+        ...[
+          {
+            name: "EMBEDDING_DIM",
```

Comment:
All providers will have an embedding and that embedding has a dimension so this must be outside of this switch